src/historical.rs

```diff
@@ -113,10 +109,6 @@ impl Historical {
         let db = WalletDB::get_connection();
         let historical = db.collection("historical");
 
-        // We search for historical prices over a week to make sure we get
-        // data even through weekends and holidays.
-        // FIXME: this version of the mongodb driver doesn't seem to like
-        // DateTime<Utc> objects. Newer ones work, maybe bite the bullet here.
```

Comment:
This FIXME is about the .to_rfc3339() calls here. Try removing them and passing the date objects to mongo directly, does it work?